src/interceptors/XMLHttpRequest/XMLHttpRequestOverride.ts

```diff
@@ -358,6 +358,9 @@ export const createXMLHttpRequestOverride = (
               this.trigger('load')
 
               const responseHeaders = originalRequest.getAllResponseHeaders()
+              this.responseHeaders = flattenHeadersObject(
+                headersToObject(stringToHeaders(responseHeaders))
+              )
```

Comment:
I see that we have forgotten to return back the headers

Reply:
Is it necessary? We keep an internal `responseHeaders` property to get/set response headers. That's also the purpose of the same internal property on the original XHR (the one I used to get via a Symbol).

If you take a look at `[XMLHttpRequest properties](https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest#Properties) you can see that `responseHeaders` isn't one of them. Developers should retrieve response headers using the `getAllResponseHeaders` or `getResponseHeader` methods of the XHR instance.

Reply:
Yes , but how the user will retrieve headers if the requests are different? 

The user creates an instance of XMLHttpRequest that is our custom class. Our class then call the real XMLHttpRequest. The real XMLHttpRequest has headers but our instance not

Reply:
🤔 That makes sense. Sorry I've overlooked that.

Reply:
Did your new test fail without this assignment? I wonder if we're covering this behavior.

Reply:
Yes, accessing the header from `req` instead of the event function will cover that case